tests/DependencyInjection/DoctrineMongoDBExtensionTest.php

```diff
@@ -716,6 +727,61 @@ public function testAutoEncryptionMinimumODMVersion(): void
         $loader->load([$config], $container);
     }
 
+    #[DataProvider('provideLazyObjectConfigurations')]
+    public function testRegistryGetManagerForClass(array $config): void
+    {
+        $container = $this->getContainer('AttributesBundle');
+        $loader    = new DoctrineMongoDBExtension();
+        $loader->load(self::buildConfiguration($config + [
+            'connections' => [
+                'default' => [],
+            ],
+            'document_managers' => [
+                'default' => [
+                    'mappings' => [
+                        'AttributesBundle' => ['type' => 'attribute'],
+                    ],
+                ],
+            ],
+        ]), $container);
+        (new ResolveParameterPlaceHoldersPass())->process($container);
+        (new ServiceRepositoryCompilerPass())->process($container);
+        (new CreateProxyDirectoryPass())->process($container);
+
+        $container->compile();
+        $registry = $container->get('doctrine_mongodb');
+        $dm       = $container->get('doctrine_mongodb.odm.document_manager');
+
+        self::assertInstanceOf(ManagerRegistry::class, $registry);
+        self::assertInstanceOf(DocumentManager::class, $dm);
+
+        // Create a lazy object
```

Comment:
Consider a bigger comment that explains the last assertion on the DM registry is actually pertinent to testing the 5th argument (and your bug fix).

Assertion messages also work fine.